marker_adapter/src/loader.rs

```diff
@@ -40,53 +42,58 @@ fn windows_split_os_str(s: &OsStr, c: u8) -> Vec<OsString> {
     bytes.split(|v| *v == u16::from(c)).map(OsString::from_wide).collect()
 }
 
-/// This struct loads external lint crates into memory and provides a safe API
-/// to call the respective methods on all of them.
-#[derive(Default)]
-pub struct LintCrateRegistry {
-    passes: Vec<LoadedLintCrate>,
+/// A struct describing a lint crate that can be loaded
+#[derive(Debug, Clone)]
+pub struct LintCrateInfo {
+    /// The absolute path of the compiled dynamic library, which can be loaded as a lint crate.
+    pub path: PathBuf,
 }
 
-impl LintCrateRegistry {
+impl LintCrateInfo {
+    /// This function tries to load the list of [`LintCrateInfo`]s from the
+    /// [`LINT_CRATES_ENV`] environment value.
+    ///
     /// # Errors
-    /// This can return errors if the library couldn't be found or if the
-    /// required symbols weren't provided.
-    fn load_external_lib(lib_path: &OsStr) -> Result<LoadedLintCrate, LoadingError> {
-        let lib: &'static Library = Box::leak(Box::new(
-            unsafe { Library::new(lib_path) }.map_err(|_| LoadingError::FileNotFound)?,
-        ));
+    ///
+    /// This function will return an error if the value can't be read or the
+    /// content is malformed. The `README.md` of this adapter contains the
+    /// format definition.
+    pub fn list_from_env() -> Result<Vec<LintCrateInfo>, AdapterError> {
+        let Some(env_str) = std::env::var_os(LINT_CRATES_ENV) else {
+            return Err(AdapterError::LintCratesEnvUnset);
+        };
 
-        let pass = LoadedLintCrate::try_from_lib(lib)?;
+        let mut list = vec![];
+        for entry_str in split_os_str(&env_str, b';') {
+            if entry_str.is_empty() {
+                return Err(AdapterError::LintCratesEnvMalformed);
+            }
 
-        // FIXME: Create issue for lifetimes and fix dropping and pointer decl stuff
+            list.push(LintCrateInfo {
+                path: PathBuf::from(entry_str),
+            });
+        }
 
-        Ok(pass)
+        Ok(list)
```

Comment:
Yeah, these for loops often just arise, when I draft code and refactor it over and over. :+1:

Reply:
> Also maybe this should use the [doc.rust-lang.org/stable/std/env/fn.split_paths.html](https://doc.rust-lang.org/stable/std/env/fn.split_paths.html) format?

Yep, just never knew about that one. I expect that this format will change soon, to also include the lint crate name for #201, but for now we can use that function :+1: And then it would probably be smart to use Serde to serialize and deserialize these entries.

cc: @NathanFrasier since this references #201